corehq/apps/users/validation.py

```diff
@@ -17,7 +17,7 @@ def validate_mobile_username(username, domain):
 
     _validate_complete_username(username, domain)
 
-    if not is_username_available(username):
+    if not is_username_available(username) and is_unique:
```

Comment:
`is_username_available` hits the db, so you should check if `is_unique` is true first otherwise it isn't worth it.
```suggestion
    if is_unique and not is_username_available(username):
```